src/components/Button/Button.tsx

```diff
@@ -70,7 +70,7 @@ export class Button extends React.Component<IButtonProps, IButtonState> implemen
         {
           'aria-label': ariaLabel,
           'aria-labelledby': ariaLabel ? null : labelId,
-          'aria-describedby': ariaDescription ? ariaDescriptionId : description ? descriptionId : null,
+          'aria-describedby': ariaDescription ? ariaDescriptionId : description ? descriptionId : nativeProps['aria-describedby'] ? nativeProps['aria-describedby'] : null,
```

Comment:
I'd have to break it out into a multi step set of conditionals, but it would be easier to read/document.